master/internal/db/postgres_tasks.go

```diff
@@ -192,11 +192,17 @@ func (db *PgDB) UpdateAllocationState(a model.Allocation) error {
 // CloseOpenAllocations finds all allocations that were open when the master crashed
 // and adds an end time.
 func (db *PgDB) CloseOpenAllocations() error {
+	cluster_heartbeat, err := db.GetClusterHeartBeat()
+
+	if err != nil {
+		return err
+	}
+
 	if _, err := db.sql.Exec(`
 UPDATE allocations
-SET end_time = current_timestamp AT TIME ZONE 'UTC'
+SET end_time = $1
 WHERE end_time IS NULL
-`); err != nil {
+`, cluster_heartbeat); err != nil {
```

Comment:
I would prefer something like
```
UPDATE allocations SET end_time = cluster_heartbeat FROM cluster_id;
```
and dropping the code above this, with the advantages mainly being `GetClusterHeartBeat` becomes unused and less code is always better (and this becomes much less code), this is 1 db call instead of 2 (but this is not a big deal since it's every 10m, but applies elsewhere so it makes sense to try to be consistent).

Reply:
Yes! I'm not sure why I did it this way in the first place. Getting directly from the cluster_id makes a lot more sense.

Reply:
I changed the sql query to this: UPDATE allocations
SET end_time = cluster_heartbeat FROM cluster_id
WHERE end_time IS NULL. We need the end_time is NULL bit to ensure that only the open allocations end_time is set to cluster heartbeat, correct?

Reply:
And I got rid fo the GetClusterHeartBeat method from postgres_cluster.go